core/templates/pages/community-dashboard-page/translation-language-selector/translation-language-selector.component.spec.ts

```diff
@@ -0,0 +1,162 @@
+  // eslint-disable-next-line max-len
+  'pages/community-dashboard-page/services/contribution-opportunities-backend-api.service.ts';
+import { FeaturedTranslationLanguageObjectFactory } from
+  'domain/opportunity/FeaturedTranslationLanguageObjectFactory';
+
+/**
+ * @fileoverview Unit tests for the translation language selector component.
+ */
+
+describe('Translation language selector', () => {
+  let component: TranslationLanguageSelectorComponent;
+  let fixture: ComponentFixture<TranslationLanguageSelectorComponent>;
+
+  let featuredTranslationLanguageObjectFactory =
+    new FeaturedTranslationLanguageObjectFactory();
+  let featuredLanguages = [
+    featuredTranslationLanguageObjectFactory.createFromBackendDict({
+      language_code: 'fr',
+      explanation: 'Partnership with ABC'
+    }),
+    featuredTranslationLanguageObjectFactory.createFromBackendDict({
+      language_code: 'de',
+      explanation: 'Partnership with CBA'
+    })
+  ];
+
+  let contributionOpportunitiesBackendApiServiceStub:
+    Partial<ContributionOpportunitiesBackendApiService> = {
+      fetchFeaturedTranslationLanguages: () =>
+        Promise.resolve(featuredLanguages)
+    };
+
+  let clickDropdown: () => void;
+  let getDropdownOptionsContainer: () => HTMLElement;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [TranslationLanguageSelectorComponent],
+      providers: [{
+        provide: ContributionOpportunitiesBackendApiService,
+        useValue: contributionOpportunitiesBackendApiServiceStub
+      }]
+    })
+      .compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(TranslationLanguageSelectorComponent);
+    component = fixture.componentInstance;
+    component.activeLanguageCode = 'en';
+    fixture.detectChanges();
+  });
+
+  beforeEach(() => {
+    clickDropdown = () => {
+      fixture.debugElement.nativeElement
+        .querySelector('.oppia-translation-language-selector-inner-container')
+        .click();
+      fixture.detectChanges();
+    };
+
+    getDropdownOptionsContainer = () => {
+      return fixture.debugElement.nativeElement.querySelector(
+        '.oppia-translation-language-selector-dropdown-container');
+    };
+  });
+
+  it('should correctly initialize languageIdToDescription map', () => {
+    expect(component.languageIdToDescription.en).toEqual('English');
+    expect(component.languageIdToDescription.fr).toEqual('French');
+  });
+
+  it('should correctly fetch featured languages', async(() => {
+    fixture.whenStable().then(() => {
+      fixture.detectChanges();
+      expect(component.featuredLanguages).toEqual(featuredLanguages);
+    });
+  }));
+
+  it('should correctly initialize dropdown activeLanguageCode', () => {
+    const dropdown = (fixture.nativeElement
+      .querySelector('.oppia-translation-language-selector-inner-container'));
+
+    expect(dropdown.firstChild.textContent.trim()).toEqual('English');
```

Comment:
```suggestion
    expect(dropdown.firstChild.textContent.trim()).toBe('English');
```

Here and elsewhere. Just to clarify, the `toBe` method is to check primitive values as number, string, boolean. The `toEqual` method checks reference values, as object and arrays,

Reply:
Done